copy/copy.go

```diff
@@ -157,6 +158,17 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 		}
 	}()
 
+	c, err := src.OCIConfig()
+	if err != nil {
+		return err
+	}
+	osErr := fmt.Errorf("image operating system %q cannot be used on %q", c.OS, runtime.GOOS)
+	if runtime.GOOS == "windows" && c.OS == "linux" {
+		return osErr
+	} else if runtime.GOOS != "windows" && c.OS == "windows" {
+		return osErr
+	}
```

Comment:
BTW please consider moving the logic out into a separate function to try and keep `copy.Image` from growing _too_ quickly, especially if it had to be much longer than this. (Absolutely non-blocking.)